pkgpanda/http/__init__.py

```diff
@@ -0,0 +1,229 @@
+"""Pkgpanda HTTP API"""
+
+import http.client
+import logging
+import os
+import sys
+
+from flask import Flask, current_app, jsonify, request
+
+from pkgpanda import Install, Repository, actions
+from pkgpanda.exceptions import (PackageConflict, PackageError,
+                                 PackageNotFound, ValidationError)
+
+
+empty_response = ('', http.client.NO_CONTENT)
+
+
+def package_listing_response(package_ids):
+    return jsonify(sorted(package_ids))
+
+
+def package_response(package_id, repository):
+    try:
+        package = repository.load(package_id)
+    except ValidationError:
+        response = (
+            error_response("Invalid package ID: {}".format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageNotFound:
+        response = (
+            error_response('Package {} not found.'.format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageError:
+        error_message = 'Unable to load package {}.'.format(package_id)
+        logging.exception(error_message)
+        response = (
+            error_response(error_message),
+            http.client.INTERNAL_SERVER_ERROR,
+        )
```

Comment:
If you're referring to the duplicated 404 in response to `PackageNotFound` here and in `remove_package()`, I can DRY that up by generating the response from a `package_404()` function. We'd still want the view to decide which error to send to the client, since different views may send different errors in response to that exception, or possibly (certainly in the case of the CLI) no error at all.